airflow/jobs/scheduler_job.py

```diff
@@ -1633,6 +1634,9 @@ def _orphan_unreferenced_datasets(self, session: Session = NEW_SESSION) -> None:
                 )
             )
         )
+        orphaned_datasets_count = 0
         for dataset in orphaned_dataset_query:
             self.log.info("Orphaning unreferenced dataset '%s'", dataset.uri)
             dataset.is_orphaned = expression.true()
+            orphaned_datasets_count += 1
+        Stats.incr("dataset.orphaned_datasets", count=orphaned_datasets_count)
```

Comment:
```suggestion
        Stats.incr("dataset.orphaned", count=orphaned_datasets_count)
```

Same comment here, "dataset" is already in the metric name so would deduplicate.